NCEP/gdas_utility.py

```diff
@@ -44,7 +47,57 @@ def __init__(self, start_datetime, end_datetime, output_directory=None, download
 
         # List of file formats to download
         self.file_formats = ['0p25.f000', '0p25.f006'] # , '0p25.f001'
+    
+    def s3bucket(self, date_str, time_str, local_directory):
+        # Construct the S3 prefix for the directory
+        s3_prefix = f"{self.root_directory}.{date_str}/{time_str}/"
+        # List objects in the S3 directory
+        s3_objects = self.s3.list_objects_v2(Bucket=self.bucket_name, Prefix=s3_prefix)
+
+        # Filter objects based on the desired formats
+        for obj in s3_objects.get('Contents', []):
+            obj_key = obj['Key']
+            for file_format in self.file_formats:
+                if obj_key.endswith(f'.{file_format}'):
+
+                    # Define the local file path
+                    local_file_path = os.path.join(local_directory, os.path.basename(obj_key))
+
+                    # Download the file from S3 to the local path
+                    self.s3.download_file(self.bucket_name, obj_key, local_file_path)
+                    print(f"Downloaded {obj_key} to {local_file_path}")
+    
+    def nomads(self, date_str, time_str, local_directory):
+        # Construct the URL for the data directory
+        gdas_url = f"https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/{self.root_directory}.{date_str}/{time_str}/atmos/"
```

Comment:
What is "f" here before the url?

Reply:
in python it means these `{self.root_directory}.{date_str}/{time_str}` should be imputed and these are not strings, it tells python we have variables in the string